File: ipman.shared/Entity/Join/SiteAccountUserAccount.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ipman.shared.Entity.Join
{
    public class SiteAccountUserAccount: EntityBase
    {
        public Guid SiteAccountID { get; set; }
        public SiteAccount SiteAccount { get; set; }
        public Guid UserAccountID { get; set; }
        public UserAccount UserAccount { get; set; }
        public Guid RoleID { get; set; }
        public bool IsActive { get; set; }
        public bool IsMemberOfAllDepartments { get; set; }
        public DateTime LastLoginUTC { get; set; }
        [NotMapped]
        public Role Role => Role.Find(RoleID);
    }
}

File: ipman/ClientApp/entity/join/SiteAccountUserAccount.ts
import SiteAccount from "../SiteAccount";
import UserAccount from "../UserAccount";
import Role from "../Role";

export default interface SiteAccountUserAccount {
    SiteAccountID: string;
    SiteAccount?: SiteAccount;
    UserAccountID: string;
    UserAccount?: UserAccount;
    RoleID: string;
    IsActive: boolean;
    IsMemberOfAllDepartments: boolean;
    LastLoginUTC: Date | string;
    Role: Role;
}